### PR TITLE
Source control sample

### PR DESCRIPTION
**Pull request by eskatos**
See #550

### CODE REVIEW

File: samples/source-control/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    cd sample
+    ./gradlew run
+
+And check compilation dependencies with:
+
+    cd sample
+    ./gradlew dependencies --configuration compileClasspath
+
+See [sample/settings.gradle.kts](./sample/settings.gradle.kts).
+
```
Comment (JLLeitschuh) [inline samples/source-control/README.md]: This really doesn't help explain what this is demoing.

I don't really understand what you are trying to demo here. Composite builds? Something else? Can you add some more details to this readme to explain?

File: samples/source-control/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+source-control
+==============
+
+A project that depend on external sources:
```
Comment (bamboo) [inline samples/source-control/README.md]: :x: _... depends on ..._

File: samples/source-control/external/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        inputs.dir("src")
+        inputs.files("build.gradle.kts", "settings.gradle.kts")
+
+        val gitRepoDir = file("$buildDir/git-repo")
+        outputs.dir(gitRepoDir)
+
+        doLast {
+            delete(gitRepoDir)
+            delete(temporaryDir)
```
Comment (bamboo) [inline samples/source-control/external/build.gradle.kts]: :x: Is `delete(temporaryDir)` really necessary?

File: samples/source-control/external/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+            val bare = Git.init()
```
Comment (bamboo) [inline samples/source-control/external/build.gradle.kts]: 👍 Very nice!

File: samples/README.md [hunk 5]
```diff
@@ -31,6 +31,7 @@ The Gradle projects in this directory demonstrate typical use cases with and fea
  - [`project-properties`](./project-properties): demonstrates project property access via [delegated properties](https://kotlinlang.org/docs/reference/delegated-properties.html)
  - [`project-with-buildSrc`](./project-with-buildSrc): demonstrates a single-project build with custom build logic in `buildSrc`
  - [`provider-properties`](./provider-properties): demonstrates usage of lazily evaluated properties to [map extension properties to task properties](https://docs.gradle.org/4.0-milestone-2/userguide/custom_plugins.html#sec:mapping_extension_properties_to_task_properties)
+ - [`source-control`](./source-control): demonstrates how to use external source dependencies
```
Comment (bamboo) [inline samples/README.md]: :x: Link to the [source dependencies epic](https://github.com/gradle/gradle-native/issues/42)? At least until we have a more authoritative.

File: samples/source-control/README.md (hunk 2 above)
Comment (bamboo) [reply samples/source-control/README.md]: ⭕️ Maybe something like:

> A project that depends on external sources using the incubating support for [source dependencies](https://github.com/gradle/gradle-native/issues/42):

File: samples/source-control/external/build.gradle.kts (hunk 3 above)
Comment (eskatos) [reply samples/source-control/external/build.gradle.kts]: Yes, otherwise jgit throws.

File: samples/source-control/README.md (hunk 1 above)
Comment (eskatos) [reply samples/source-control/README.md]: Done, see https://github.com/gradle/gradle-native/issues/42

File: samples/README.md (hunk 5 above)
Comment (eskatos) [reply samples/README.md]: Done

File: samples/source-control/README.md (hunk 2 above)
Comment (eskatos) [reply samples/source-control/README.md]: Done, thanks

File: samples/source-control/README.md (hunk 1 above)
Comment (JLLeitschuh) [reply samples/source-control/README.md]: Thanks, I had no idea that this was related to building native code.

**Pull request merged**